pandas/core/window.py

```diff
@@ -0,0 +1,1339 @@
+
+            # if we have a string function name, wrap it
+            if isinstance(func, compat.string_types):
+                if not hasattr(algos, func):
+                    raise ValueError("we do not support this function algos.{0}".format(func))
+
+                cfunc = getattr(algos, func)
+                def func(arg, window, min_periods=None):
+                    minp = check_minp(min_periods, window)
+                    return cfunc(arg, window, minp, **kwargs)
+
+            # calculation function
+            if center:
+                offset = _offset(window, center)
+                additional_nans = np.array([np.NaN] * offset)
+                def calc(x):
+                    return func(np.concatenate((x, additional_nans)),
+                                window, min_periods=self.min_periods)
+            else:
+                def calc(x):
+                    return func(x,window, min_periods=self.min_periods)
+
+            if values.ndim > 1:
+                result = np.apply_along_axis(calc, self.axis, values)
+            else:
+                result = calc(values)
+
+            if center:
+                result = self._center_window(result, window)
+
+            results.append(result)
+
+        return self._wrap_results(results, blocks, obj)
+
+class _Rolling_and_Expanding(_Rolling):
+
+    _shared_docs['count'] = """%(name)s count of number of non-NaN observations inside provided window."""
+    def count(self):
+        obj = self._convert_freq()
+        window = self._get_window()
+        window = min(window, len(obj)) if not self.center else window
+        try:
+            converted = np.isfinite(obj).astype(float)
+        except TypeError:
+            converted = np.isfinite(obj.astype(float)).astype(float)
+        result = self._constructor(converted,
+                                   window=window,
+                                   min_periods=0,
+                                   center=self.center).sum()
+
+        result[result.isnull()] = 0
+        return result
+
+    _shared_docs['apply'] = dedent("""
+    %(name)s function apply
+
+    Parameters
+    ----------
+    func : function
+    Must produce a single value from an ndarray input
+    *args and **kwargs are passed to the function""")
+
+    def apply(self, func, args=(), kwargs={}):
+        _level = kwargs.pop('_level',None)
+        window = self._get_window()
+        offset = _offset(window, self.center)
+        def f(arg, window, min_periods):
+            minp = _use_window(min_periods, window)
+            return algos.roll_generic(arg, window, minp, offset, func, args, kwargs)
+
+        return self._apply(f, center=False)
+
+    _shared_docs['sum'] = """%(name)s sum"""
+    def sum(self):
+        return self._apply('roll_sum')
+
+    _shared_docs['max'] = dedent("""
+    %(name)s maximum
+
+    Parameters
+    ----------
+    how : string, default max
+    Method for down- or re-sampling""")
```

Comment:
the 'Method ...' line should be indented compared to 'how : string ...' (as it is the explanation part of the parameter)